{
  final Throwable failure=new RuntimeException();
  final Collector<Throwable> wrappedFailure=collector();
  context.checking(new Expectations(){
{
      allowing(task).getEnabled();
      will(returnValue(true));
      one(spec).isSatisfiedBy(task);
      will(throwException(failure));
      one(state).executed(with(notNullValue(GradleException.class)));
      will(collectTo(wrappedFailure));
    }
  }
);
  executer.execute(task,state);
  GradleException exception=(GradleException)wrappedFailure.get();
  assertThat(exception.getMessage(),equalTo("Could not evaluate onlyIf predicate for <task>."));
  assertThat(exception.getCause(),sameInstance(failure));
}
